src/inference/src/dev/core_impl.cpp

```diff
@@ -276,241 +290,391 @@ ov::InferencePlugin ov::CoreImpl::GetCPPPluginByName(const std::string& pluginNa
+    auto& config_with_batch = parsed._config;
+    // if auto-batching is applicable, the below function will patch the device name and config accordingly:
+    apply_auto_batching(model, deviceName, config_with_batch);
+    clean_properties(deviceName, config_with_batch, ov::auto_batch_timeout);
+    parsed = parseDeviceNameIntoConfig(deviceName, config_with_batch);
+
+    auto plugin = get_plugin(parsed._deviceName);
     ov::SoPtr<InferenceEngine::IExecutableNetworkInternal> res;
     auto cacheManager =
-        coreConfig.getCacheConfigForDevice(parsed._deviceName, DeviceSupportsCacheDir(plugin), parsed._config)
+        coreConfig.get_cache_config_for_device(parsed._deviceName, device_supports_cache_dir(plugin), parsed._config)
             ._cacheManager;
     auto cacheContent = CacheContent{cacheManager};
-    if (!forceDisableCache && cacheManager && DeviceSupportsImportExport(plugin)) {
-        cacheContent.blobId = CalculateNetworkHash(network, parsed._deviceName, plugin, parsed._config);
+    if (cacheManager && device_supports_import_export(plugin)) {
+        cacheContent.blobId = CalculateNetworkHash(ov::legacy_convert::convert_model(model, is_new_api()),
+                                                   parsed._deviceName,
+                                                   plugin,
+                                                   parsed._config);
         bool loadedFromCache = false;
         auto lock = cacheGuard.getHashLock(cacheContent.blobId);
-        res = LoadNetworkFromCache(cacheContent, plugin, parsed._config, nullptr, loadedFromCache);
+        res = load_model_from_cache(cacheContent, plugin, parsed._config, context, loadedFromCache);
         if (!loadedFromCache) {
-            res = compile_model_impl(network, plugin, parsed._config, nullptr, cacheContent, forceDisableCache);
+            res = compile_model_impl(model, plugin, parsed._config, context, cacheContent);
         } else {
             // Temporary workaround until all plugins support caching of original model inputs
-            InferenceEngine::SetExeNetworkInfo(res._ptr, network.getFunction(), isNewAPI());
+            InferenceEngine::SetExeNetworkInfo(res._ptr, model, isNewAPI());
         }
     } else {
-        res = compile_model_impl(network, plugin, parsed._config, nullptr, cacheContent, forceDisableCache);
+        res = compile_model_impl(model, plugin, parsed._config, context, cacheContent);
     }
-    return {res._ptr, res._so};
+    return res;
+}
+ov::SoPtr<InferenceEngine::IExecutableNetworkInternal> ov::CoreImpl::compile_model(
+    ov::Plugin& plugin,
+    const std::shared_ptr<const ov::Model>& model,
+    const ov::RemoteContext& context,
+    const ov::AnyMap& config) const {
+    std::shared_ptr<ov::Model> cloned_model = model->clone();
+    ov::SoPtr<InferenceEngine::IExecutableNetworkInternal> compiled_model;
+
+    if (!is_new_api() && !std::dynamic_pointer_cast<InferenceEngine::IPluginWrapper>(plugin.m_ptr)) {
```

Comment:
Honestly I don't know the case why somebody excepts CoreImpl should know about new or old plugin API. Here it can be checked by dynamic cast to wrapper instance